lib/vmpooler/api/v1.rb

```diff
@@ -455,13 +456,14 @@ def invalid_templates(payload)
         result[params[:hostname]] = {}
 
         result[params[:hostname]]['template'] = rdata['template']
-        result[params[:hostname]]['lifetime'] = (rdata['lifetime'] || config['vm_lifetime']).to_i
+        pool_auth = pools.find {|p| p["name"] == rdata['template'] }['vm_lifetime']
+        result[params[:hostname]]['lifetime'] = (rdata['lifetime'] || pool_auth || config['vm_lifetime']).to_i
 
         if rdata['destroy']
-          result[params[:hostname]]['running'] = ((Time.parse(rdata['destroy']) - Time.parse(rdata['checkout'])) / 60 / 60).round(2)
+          result[params[:hostname]]['running'] = ((Time.parse(rdata['destroy']) - Time.parse(rdata['checkout'])) / 60 / 60).round(2).to_f
```

Comment:
We do a lot of date math throughout this file; we should move that date math into some helper functions. Probably out of scope for this PR.